### DISCUSSION TITLE
DNSDist export rules statistics as Prometheus metrics

### DISCUSSION THREAD

**Discussion by KVInventoR** [PowerDNS/pdns#13590] [Q&A]
Hi there,

I am trying to find a right approach to get metrics statistics for rules on distdist, is there any simple Prometheus way how can I do it?
Or would it be possible to extend existing Prometheus exports to support also exporting rules statistic?

Another question will be related, would it be possible to add also comments for rules? I would like to see not only domains, but also some comments.
Like this:
addAction(RegexRule("SOME_WEB_URL", COMMENT("BLOCKED BY SPAM)), SpoofAction("127.0.0.1"))

thank you

**Comment by phonedph1** [comment]
You can already do this by giving it a name to identify it.

```
addAction("google.com",DropAction(),{name="this is a name for the rule"})

$ curl -su dnsdist:dnsdist 127.0.0.1:8080/metrics | grep dnsdist_rule_hits
# HELP dnsdist_rule_hits Number of hits of that rule
# TYPE dnsdist_rule_hits counter
dnsdist_rule_hits{id="this is a name for the rule"} 6
```

If you don't give a name you get the UUID.

```
> showRules({showUUIDs=true})
#   Name                           UUID                                   Cr. Order   Matches Rule                                                     Action
[...]
3                                  edd2f3da-25b1-4f73-8278-2b8a013aa4bf           4         2 qname in 2google.com.                                    drop

dnsdist_rule_hits{id="edd2f3da-25b1-4f73-8278-2b8a013aa4bf"} 2
```

I personally run dnsdist with a bit of an older patch so the /metrics endpoint also includes the rule itself and action itself, but this is very un-prometheus (but more useful to me). https://gist.github.com/phonedph1/16e4c617b5e05be6a86ad43f38505baa
